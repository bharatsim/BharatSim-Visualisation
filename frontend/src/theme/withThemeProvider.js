import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
