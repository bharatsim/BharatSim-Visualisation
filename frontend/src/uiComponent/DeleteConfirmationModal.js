import React from 'react';
import PropTypes from 'prop-types';

import Dialog from '@material-ui/core/Dialog';
import { makeStyles, withStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import MuiDialogContent from '@material-ui/core/DialogContent';
import ErrorButton from './ErrorButton';
import { DialogActions, DialogTitle } from './Modal';

const dialogStyles = makeStyles((theme) => ({
  paper: {
    width: theme.spacing(121),
  },
}));
const DialogContent = withStyles((theme) => ({
  root: {
    margin: theme.spacing(0, 4),
    padding: theme.spacing(0, 0, 4, 0),
  },
}))(MuiDialogContent);

function DeleteConfirmationModal({ open, handleClose, title, children, deleteAction }) {
  const dialogClasses = dialogStyles();
  const { onDelete, name, dataTestId } = deleteAction;
  return (
    <Dialog
      onClose={handleClose}
      aria-labelledby="customized-dialog-title"
      open={open}
      classes={dialogClasses}
    >
      <DialogTitle id="customized-dialog-title" onClose={handleClose}>
        {title}
      </DialogTitle>
      <DialogContent>
        <Box px={2}>{children}</Box>
      </DialogContent>

      <DialogActions>
        <ErrorButton data-testid={dataTestId} autoFocus onClick={onDelete} variant="contained">
          {name}
        </ErrorButton>
      </DialogActions>
    </Dialog>
  );
}

DeleteConfirmationModal.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  deleteAction: PropTypes.shape({
    name: PropTypes.string.isRequired,
    dataTestId: PropTypes.string.isRequired,
    onDelete: PropTypes.func.isRequired,
  }).isRequired,
};

export default DeleteConfirmationModal;
