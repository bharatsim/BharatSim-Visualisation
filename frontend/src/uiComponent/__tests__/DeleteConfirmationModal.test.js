import React from 'react';
import { render, within } from '@testing-library/react';

import { fireEvent } from '@testing-library/dom';
import DeleteConfirmationModal from '../DeleteConfirmationModal';
import withThemeProvider from '../../theme/withThemeProvider';

describe('<DeleteConfirmationModal />', () => {
  const DeleteConfirmationModalWithTheme = withThemeProvider(DeleteConfirmationModal);

  it('should match snapshot', () => {
    render(
      <DeleteConfirmationModalWithTheme
        open
        handleClose={jest.fn()}
        title="Modal"
        deleteAction={{ name: 'Delete', onDelete: jest.fn(), dataTestId: 'test' }}
      >
        Hello this is modal
      </DeleteConfirmationModalWithTheme>,
    );

    expect(document.querySelector('.MuiPaper-root')).toMatchSnapshot();
  });

  it('should call handle close callback on click of close icon button', () => {
    const handleClose = jest.fn();
    render(
      <DeleteConfirmationModalWithTheme
        open
        handleClose={handleClose}
        title="Modal"
        deleteAction={{ name: 'Delete', onDelete: jest.fn(), dataTestId: 'test' }}
      >
        Hello this is modal
      </DeleteConfirmationModalWithTheme>,
    );

    const modal = within(document.querySelector('.MuiPaper-root'));

    const closeIconButton = modal.getByTestId('button-icon-close');

    fireEvent.click(closeIconButton);

    expect(handleClose).toHaveBeenCalled();
  });

  it('should call action callback on click of first action button', () => {
    const onDelete = jest.fn();
    render(
      <DeleteConfirmationModalWithTheme
        open
        handleClose={jest.fn()}
        title="Modal"
        das
        deleteAction={{ name: 'Delete', onDelete, dataTestId: 'testId' }}
      >
        Hello this is modal
      </DeleteConfirmationModalWithTheme>,
    );

    const modal = within(document.querySelector('.MuiPaper-root'));
    const deleteButton = modal.getByTestId('testId');

    fireEvent.click(deleteButton);

    expect(onDelete).toHaveBeenCalled();
  });
});
