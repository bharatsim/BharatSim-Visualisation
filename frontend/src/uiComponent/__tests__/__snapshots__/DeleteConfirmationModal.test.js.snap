// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteConfirmationModal /> should match snapshot 1`] = `
<div
  aria-labelledby="customized-dialog-title"
  class="MuiPaper-root MuiDialog-paper makeStyles-paper-1 MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
  role="dialog"
>
  <div
    class="MuiDialogTitle-root Component-root-2"
    id="customized-dialog-title"
  >
    <div
      class="MuiBox-root MuiBox-root-4 Component-titleContainer-3"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        Modal
      </h5>
      <button
        aria-label="close"
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="button-icon-close"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiDialogContent-root WithStyles(ForwardRef(DialogContent))-root-5"
  >
    <div
      class="MuiBox-root MuiBox-root-6"
    >
      Hello this is modal
    </div>
  </div>
  <div
    class="MuiDialogActions-root WithStyles(ForwardRef(DialogActions))-root-7 MuiDialogActions-spacing"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained-9 Mui-focusVisible Mui-focusVisible"
      data-testid="test"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Delete
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
          style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
          />
        </span>
      </span>
    </button>
  </div>
</div>
`;
